opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java

```diff
@@ -281,5 +281,33 @@ void testPaymentInitializationUsingEmbedded(Approach expectedApproach) {
                 .fintech_calls_payment_activation_for_current_authorization_id()
                 .fintech_calls_payment_status();
     }
+
+    @ParameterizedTest
+    @EnumSource(Approach.class)
+    void testPaymentInitializationUsingEmbeddedWithPhotoOtp(Approach expectedApproach) {
+        given()
+                .embedded_mock_of_sandbox_for_max_musterman_payments_running()
+                .set_default_preferred_approach()
+                .preferred_sca_approach_selected_for_all_banks_in_opba(expectedApproach)
+                .rest_assured_points_to_opba_server()
+                .user_registered_in_opba_with_credentials(OPBA_LOGIN, OPBA_PASSWORD);
+
+        when()
+                .fintech_calls_initiate_payment_for_max_musterman()
+                .and()
+                .user_logged_in_into_opba_as_opba_user_with_credentials_using_fintech_supplied_url(OPBA_LOGIN, OPBA_PASSWORD)
+                .and()
+                .user_max_musterman_provided_initial_parameters_to_make_payment()
+                .and()
+                .user_max_musterman_provided_password_to_embedded_authorization()
+                .and()
+                .user_max_musterman_selected_sca_challenge_type_photo_otp_to_embedded_authorization()
```

Comment:
There should be a check to verify that PhotoTAN data is available to frontend (GetAuthorizationState call)

Reply:
Done